src/Microsoft.AspNetCore.Blazor/Routing/NavLink.cs

```diff
@@ -100,19 +100,20 @@ private bool ShouldMatch(string currentUriAbsolute)
 
         private void Render(RenderTreeBuilder builder)
         {
-            builder.OpenElement(0, "a");
+            var i = 0;
+            builder.OpenElement(i++, "a");
 
             // Set "active" class dynamically
-            builder.AddAttribute(0, "class", CombineWithSpace(_cssClass, _isActive ? "active" : null));
+            builder.AddAttribute(i++, "class", CombineWithSpace(_cssClass, _isActive ? "active" : null));
 
             // Pass through all other attributes unchanged
-            foreach (var kvp in _allAttributes.Where(kvp => kvp.Key != "class"))
+            foreach (var kvp in _allAttributes.Where(kvp => kvp.Key != "class" && kvp.Key != RenderTreeBuilder.ChildContent))
```

Comment:
This was just a bug, the dom would contain an attribute for the child content with its type name as the value.

Reply:
Ah yes, I think I saw that at some point. Glad you squashed that one!